packages/big-design/src/components/Tabs/spec.tsx

```diff
@@ -0,0 +1,104 @@
+import { fireEvent, render } from '@testing-library/react';
+import 'jest-styled-components';
+import React from 'react';
+
+import { Tabs, TabItem } from './Tabs';
+
+const items: TabItem[] = [{ id: 'tab1', title: 'Tab 1' }, { id: 'tab2', title: 'Tab 2' }];
+
+test('render Tabs', () => {
+  const { container } = render(<Tabs items={items} />);
+
+  expect(container.firstChild).toMatchSnapshot();
+});
+
+test('render Tabs with disabled item', () => {
+  const disabledItems: TabItem[] = [...items, { id: 'tab3', title: 'Tab 3', disabled: true }];
+  const { container } = render(<Tabs items={disabledItems} />);
+
+  expect(container.firstChild).toMatchSnapshot();
+});
+
+test('Tabs has a role', () => {
+  const { getByRole } = render(<Tabs items={items} />);
+
+  expect(getByRole('tablist')).toBeInTheDocument();
+});
+
+test('Tab items have a role', () => {
+  const { getAllByRole } = render(<Tabs items={items} />);
+
+  expect(getAllByRole('tab').length).toBe(2);
+});
+
+test('active tab has a tabindex', () => {
+  const activeTab = 'tab2';
+  const { getAllByRole } = render(<Tabs activeTab={activeTab} items={items} />);
+
+  const tabs = getAllByRole('tab');
+
+  tabs.forEach(tab => {
+    if (tab.id === activeTab) {
+      expect(tab.tabIndex).toBe(-1);
+    } else {
+      expect(tab.tabIndex).toBe(0);
+    }
+  });
+});
+
+test('onTabClick is called', () => {
+  const onClick = jest.fn();
+  const { getByText } = render(<Tabs items={items} onTabClick={onClick} />);
+  const trigger = getByText('Tab 2');
+
+  fireEvent.click(trigger);
+
+  expect(onClick).toHaveBeenCalled();
+});
+
+test('active tab changes on tab click', () => {
+  let activeTab = 'tab1';
+  const setActiveTab = jest.fn((tabId: string) => (activeTab = tabId));
+
+  const { getByText, rerender } = render(<Tabs activeTab={activeTab} items={items} onTabClick={setActiveTab} />);
+
+  let trigger = getByText('Tab 2');
+
+  expect(activeTab).toBe('tab1');
+
+  fireEvent.click(trigger);
+
+  expect(setActiveTab).toHaveBeenCalledTimes(1);
```

Comment:
🍹 You could change these to use a vanilla `jest.fn` and `toHaveBeenCalledWith`. That way you don't have to manually set `activeTab` and check for its value.